addons/base_iban/static/src/components/iban_widget/iban_widget.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { CharField } from "@web/views/fields/char/char_field";
+import { useDebounced } from "@web/core/utils/timing";
+import { useService } from "@web/core/utils/hooks";
+
+const { useState, useEffect } = owl;
+
+export class IbanWidget extends CharField {
+    setup() {
+        super.setup();
+        this.state = useState({
+            isValidIBAN: null,
+        });
+        this.orm = useService("orm");
+        this.validateIbanDebounced = useDebounced(async (ev) => {
+            const iban = ev.target.value;
+            await this.validateIban(iban);
+        }, 400);
+        // initiate isValidIBAN when edition starts
+        useEffect(() => {
+            this.validateIban(this.props.value);
+        }, () => [this.props.readonly]);
```

Comment:
this will execute when switching from edit mode to readonly too

Reply:
Do you think I should add a condition? I have the feeling that it's not really a problem. Maybe the ORM call should be avoided then.